src/OrchardCore.Modules/OrchardCore.Setup/Filters/AutoSetupStartupFilter.cs

```diff
@@ -0,0 +1,159 @@
+                        else
+                        {
+                            var shellSettings = scope.ServiceProvider.GetRequiredService<ShellSettings>();
+                            var setupService = scope.ServiceProvider.GetRequiredService<ISetupService>();
+
+                            var setupContext = GetSetupContext(options, setupService, shellSettings);
+
+                            if (options.CreateDatabase)
+                                CreateDatabase(scope, options, setupContext);
+
+                            setupService.SetupAsync(setupContext)
+                                .GetAwaiter()
+                                .GetResult();
+
+                            if (setupContext.Errors.Count == 0)
+                            {
+                                _logger.LogInformation(T["AutoSetup succesfully provisioned the site, redirecting"]);
+                                IHttpContextAccessor accessor = scope.ServiceProvider.GetRequiredService<IHttpContextAccessor>();
+                                accessor.HttpContext.Response.Redirect("/");
+                            }
+                            else
+                            {
+                                var stringBuilder = new StringBuilder();
+                                foreach (var error in setupContext.Errors)
+                                {
+                                    stringBuilder.Append($"{error.Key} : '{error.Value}'");
+                                }
+                                _logger.LogError(T["AutoSetup failed with errors: {0}", stringBuilder.ToString()]);
+                            }
+                        }
+                    }
+                }
+            }
+
+            return next;
+        }
+
+        private SetupContext GetSetupContext(AutoSetupOptions options, ISetupService setupService, ShellSettings shellSettings)
+        {
+            var recipes = setupService.GetSetupRecipesAsync()
+                .GetAwaiter()
+                .GetResult();
+
+            var recipe = recipes.SingleOrDefault(r => r.Name == options.RecipeName);
+
+            var setupContext = new SetupContext()
+            {
+                AdminEmail = options.AdminEmail,
+                AdminPassword = options.AdminPassword,
+                AdminUsername = options.AdminUsername,
+                DatabaseConnectionString = options.DatabaseConnectionString,
+                DatabaseProvider = options.DatabaseProvider,
+                DatabaseTablePrefix = options.DatabaseTablePrefix,
+                SiteName = options.SiteName,
+                Recipe = recipe,
+                SiteTimeZone = options.SiteTimeZone,
+                Errors = new Dictionary<string, string>(),
+                ShellSettings = shellSettings
+            };
+
+            return setupContext;
+        }
+
+        private void CreateDatabase(IServiceScope scope, AutoSetupOptions options, SetupContext setupContext)
```

Comment:
Don't create the database (for now). We don't support it otherwise, we shouldn't support it here only either. Let users use sqlite or table prefix instead.

Reply:
I've commented it out.

Reply:
I think you should remove the code and commented out code for now.